### PR TITLE
Enhance triage options

### PR DESCRIPTION
**Pull request by cpaelzer**
Hi Josh,
as I said I'd like to filter out those we are already subscribed.
But while at it a few more nice additions came to my mind that I added as well.
Please review and merge - feel free to discuss on IRC if anything doesn't fit your expectation.

### CODE REVIEW

**Comment by cpaelzer** [comment]
Ah - a note on the number of bugs.
I think this number is better than what we had before.

TL;DR
our old counting would count a bug with two Ubuntu tasks and a linked Debian bug as 3, this now only counts the Ubuntu bugs.

[13:23] <cpaelzer> hi, is there an obvious reason I'm overlooking why https://bugs.launchpad.net/~ubuntu-server/+subscribedbugs lists 308 hits
[13:24] <cpaelzer> but using LP api like this example http://paste.ubuntu.com/23215437/ doesn't sum up to 308 in a similar way ?
[13:25] <cpaelzer> there are a few combos which add up to 308, but they all have Fix releases, which the web view has not
[13:25] <wgrant> cpaelzer: That script is only searching for tasks in Ubuntu.
[13:25] <cpaelzer> uh, you mean we might really have some outside
[13:25] <cpaelzer> I didn't consider that yet
[13:25] <cpaelzer> let me check
[13:25] <wgrant> And the web UI by default excludes closed statuses (Invalid, Won't Fix, Expired, Opinion, Fix Released)
[13:25] <cpaelzer> thanks wgrant I knew it might be something obvious
[13:26] <wgrant> cpaelzer: Many of the bugs the team is subscribed to will have tasks on other projects or distributions in addition to the Ubuntu task.
[13:27] <cpaelzer> if a bug would have a Ubuntu tasks for Dev and Xenial and a linked Debian bug, it would still appear once in the web UI right?
[13:28] <wgrant> cpaelzer: No, bug listings, like the searchTasks API method, list tasks, not bugs.
[13:28] <cpaelzer> so they would show up twice, I'll find an exmaple
[13:28] <wgrant> You'll see dupes if you order that listing by Number.
[13:28] <cpaelzer> thanks wgrant
[13:28] <wgrant> eg. bug #1624672
[13:28] <ubot5`> bug 1624672 in dbconfig-common (Ubuntu) "Fix missing backticks in MySQL "DROP DATABASE" statement" [Low,New] https://launchpad.net/bugs/1624672
[13:29] <cpaelzer> great, all fits together now

**Comment by powersj** [comment]
@cpaelzer this looks great! Love how you have extended this. Can now be used by more and for more than just our triage list.

I am going to run some tests against it to make sure we are still returning the right numbers for our server triage. Once verified, I will merge in!

File: ubuntu_server_triage.py [hunk 1]
```diff
@@ -147,10 +181,18 @@ def main(start, end=None):
                         '(e.g. 2016-07-31)')
     PARSER.add_argument('-d', '--debug', action='store_true',
                         help='debug output')
+    PARSER.add_argument('-o', '--open', action='store_true',
+                        help='open in web browser')
+    PARSER.add_argument('-n', '--lpname', default='ubuntu-server',
+                        help='specify the launchpad name to search for')
+    PARSER.add_argument('-b', '--bugsubscriber', action='store_true',
+                        help=('filter name as bug subscriber (default would '
+                              'be structural subscriber'))
```
Comment (powersj) [inline ubuntu_server_triage.py]: What is a structural subscriber?

File: ubuntu_server_triage.py [hunk 2]
```diff
@@ -132,13 +164,15 @@ def main(start, end=None):
...
-    print_bugs(bugs)
+    report_current_backlog(lpname)
+    bugs = create_bug_list(start, end, lpname, bugsubscriber)
+    print_bugs(bugs, open_in_browser)
 
 
 if __name__ == '__main__':
     PARSER = argparse.ArgumentParser()
     PARSER.add_argument('start_date',
+                        nargs='?',
```
Comment (powersj) [inline ubuntu_server_triage.py]: I'm mixed on this one. I love the ability to search for all bugs, however as the "default" behavior with no args I hesitate as it leads to a very, very long execution time. 

What do you think about:
1) if no arguments, search for all the bugs modified today or yesterday
2) if the -a --all argument is provided, dates are ignored and we look for all bugs?

This way the default behavior would be faster and more useful?

File: README.md [hunk 3]
```diff
@@ -23,5 +23,23 @@ If two dates are given then all the bugs found on those days and between (fully
...
+## Open Bugs in Browser
+Quite commonly the triager wants to open all bugs in the browser, to read, review and manage them. Via ``open`` argument that can be done automatically.
+```
+./ubuntu_server_triage.py --open 2016-09-10 2016-09-12
+```
+
+## Launchpad Name and subscription type
+By default this searches for the structural subscription of the ubuntu-server Team.
+But depending on the use case one might overwrite the team name with `--lpname` (which can be any launchpad user, doesn't have to be a Team).
+Additionally, especially when setting a personal name it is common that the filter should be switched to check for bug subscription instead of a structural subscription which can be done via `--bugsubscriber`.
```
Comment (powersj) [inline README.md]: See my comment below, but I am confused by structural subscription. This sounds like the ones where the user has specifically added their name, as opposed to subscribed due to a list like ubuntu-server-bugs?

**Comment by cpaelzer** [comment]
Hi Josh,
please check my answers to your feedback.
Pushing the renewed changes for those that needed changes.

File: ubuntu_server_triage.py (hunk 1 above)
Comment (cpaelzer) [inline ubuntu_server_triage.py]: Well that is the field name in the search you coded yourself :-)
It is the Team or Person that is the default owner or responsible for a packet.
That seems to be the LP Term for it.

File: ubuntu_server_triage.py (hunk 2 above)
Comment (cpaelzer) [inline ubuntu_server_triage.py]: While I see your case I'd think that this isn't a tool for my grandma to use.
So I'm kind of leaning to "ok to take looooong if not with the right parms".

But, OTOH it is very easy to add a guard.
Now if run without any parms it will warn you, stopping with:
ValueError: No date set, please specify -a if you really want to search without date filter
If you add a -a it will run as before
Searching all bugs, no date filter
[...]

That should be good - If you want to make the no-parm a "auto search yesterday" I'd say feel free to do so if you really think it is needed.

File: ubuntu_server_triage.py [hunk 4]
```diff
@@ -71,37 +80,47 @@ def bug_info(bugs):
...
+
+    if bugsubscriber:
+        # direct subscriber
+        raw_bugs = project.searchTasks(modified_since=date,
+                                       bug_subscriber=team)
+    else:
+        # structural_subscriber sans already subscribed
+        mod_bugs = project.searchTasks(modified_since=date,
+                                       structural_subscriber=team)
+        already_sub_bugs = project.searchTasks(modified_since=date,
```
Comment (cpaelzer) [inline ubuntu_server_triage.py]: Execution time isn't so bad IMO, not as long as we don't start to scale with the list length too much.
But I see that you might want to see those we are already subscribed, now that we have the status added it might be less annoying to have those already subscribed in the list.

I'd suggest we make this an option which is off-by-default.
But actually while I was writing this I think it goes way better.
Avoid the clumsy extra option and just mark bugs that we are already subscribed in the overview.
I did that and the view is now simple:
not subscribed
(In Progress)
subscribed
*(In Progress)

File: ubuntu_server_triage.py [hunk 5]
```diff
@@ -53,13 +59,16 @@ def check_dates(start, end=None):
...
-def print_bugs(bugs):
+def print_bugs(bugs, open_in_browser=False):
     """
     Prints the bugs in a clean-ish format.
     """
     for bug in bugs:
         bug_url = 'https://bugs.launchpad.net/bugs/'
-        logging.info('%s%-7s - [%s] %s', bug_url, bug[0], bug[1], bug[2])
+        logging.info('%s%-7s - %-14s %-16s - %s',
+                     bug_url, bug[0], bug[3], ('[%s]' % bug[1]), bug[2])
```
Comment (cpaelzer) [inline ubuntu_server_triage.py]: We still have to easily differ e.g. packages and stated with dashes or even spaces like "In Progress" so no boundary at all might not be that readable.
I looked at your list and chose this for my next push.

https://bugs.launchpad.net/bugs/1258320 (Triaged) [libvirt] Build libvirt with Hyper-V option
